src/site/content/en/blog/streams/index.md

```diff
@@ -0,0 +1,980 @@
+---
+title: Streams—The definitive guide
+subhead: Learn how to use readable, writable, and transform streams with the Streams API.
+description: |
+  The Streams API allows JavaScript to programmatically access streams of data received over the
+  network and process them as desired.
+authors:
+  - thomassteiner
+date: 2021-01-18
+# updated: 2021-01-18
+hero: hero.jpg
+alt: A forest stream with colored fallen leaves.
+tags:
+  - blog # blog is a required tag for the article to show up in the blog.
+  - capabilities
+  - streams
+---
+
+The Streams API allows you to programmatically access streams of data received over the network
+or created by whatever means locally and
+process them with JavaScript. Streaming involves breaking a resource that you want to receive, send, or transform down
+into small chunks, and then processing these chunks bit by bit. While streaming is something
+browsers do anyway when receiving assets like HTML or videos to be shown on webpages, this
+capability has never been available to JavaScript before `fetch` with streams was introduced in 2015.
+
+{% aside %}
+  Streaming was possible with `XMLHttpRequest`, but it
+  [really was not pretty](https://gist.github.com/igrigorik/5736866).
+{% endAside %}
+
+Previously, if you wanted to process a resource of some kind (be it a video, or a text file, etc.),
+you would have to download the entire file, wait for it to be deserialized into a suitable format,
+and then process it. With streams being available to
+JavaScript, this all changes—you can now start processing raw data with JavaScript progressively as
+soon as it is available on the client-side, without needing to generate a buffer, string, or blob.
+This unlocks a number of use cases, some of which I list below:
+
+- **Video effects:** piping a readable video stream through a transform stream that applies effects
+  in real time.
+- **Data (de)compression:** piping a file stream through a transform stream that selectively
+  (de)compresses it.
+- **Image decoding:** piping an HTTP response stream through a transform stream that decodes bytes
+  into bitmap data, and then through another transform that translates bitmaps into PNGs. If
+  installed inside the `fetch` hook of a service worker, this allows you to transparently polyfill
+  new image formats like AVIF.
+
+## Core concepts
+
+Before I go into details on the various types of streams, let me introduce some core concepts first.
+
+### Chunks
+
+A chunk is a **single piece of data** that is written to or read from a stream. It can be of any
+type; streams can even contain chunks of different types. Most of the time, a chunk will not be the most atomic
```

Comment:
You understand correctly, and it is important to know, since based on this you can't make any assumptions about a chunk and need to be careful when you process them.